interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/component/defaultconfig/DefaultComponentConfigDeterminer.scala

```diff
@@ -48,22 +48,31 @@ object DefaultComponentConfigDeterminer {
     }
     SingleComponentConfig(
       params = None,
-      Some(DefaultsComponentIcon.forBuiltInComponent(info)),
+      icon = Some(DefaultsComponentIcon.forBuiltInComponent(info)),
       // TODO: move from defaultModelConfig.conf to here + convention instead of code
       docsUrl = None,
       componentGroup = Some(componentGroup),
       componentId = Some(ComponentId.forBuiltInComponent(info))
     )
   }
 
-  def forFragment(componentId: Option[ComponentId], docsUrl: Option[String]): SingleComponentConfig =
-    SingleComponentConfig(
-      params = None,
-      Some(DefaultsComponentIcon.FragmentIcon),
+  // For fragments, we don't need to return SingleComponentConfig, because this config won't merged with anything else
+  // We can just return final, ComponentUiDefinition
+  def forFragment(
+      docsUrl: Option[String],
+      translateGroupName: ComponentGroupName => Option[ComponentGroupName],
+      componentId: ComponentId,
+  ): ComponentUiDefinition = {
+    val originalGroupName = DefaultsComponentGroupName.FragmentsGroupName
+    ComponentUiDefinition(
+      originalGroupName = originalGroupName,
+      componentGroup = translateGroupName(originalGroupName)
```

Comment:
Why not to allow this? Currently we allowed to override group name for all components (also built-in).
For Built-in components we can override icon - make notice that in `BuiltInComponentsDefinitionsPreparer:25` I do combining. In this class are only default (initial) values.

Reply:
Oh, I see now. But still we allow overriding doc from fragment properties and groupName from config? There is no consistency..

Reply:
In one of changes we removed the support for configuration of all properties of fragments, based on config. We did it, because user now can configure most of things via GUI. This translation of group name is something that is parallel to `componentsUiConfig`. IMO it would be not intuitive if smb can let's say make upper case all groups but not this one. Don't you think so?

Reply:
Let's say I agree that componentGroup name config is some admin configuration..